### PR TITLE
test: add v1beta1 E2E test cases (4/)

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR adds more E2E tests, specifically concerning placement using CRPs of the PickN placement type.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A

### Special notes for your reviewer

Already features the CRP controller fix from #557.

Part of this PR overlaps with #558 + #559, which would go away after 558/559 is merged + this PR is rebased.

### CODE REVIEW

**Comment by michaelawyu** [comment]
Rebased. Note this this is expected to fail until a test fix is merged.

**Comment by michaelawyu** [comment]
This also incorporated a few style related comments unaddressed in prev. PR.

File: test/e2e/setup_test.go [hunk 1]
```diff
@@ -36,13 +36,13 @@ const (
...
-	memberCluster4Name = "kind-unhealthy-cluster"
-	memberCluster5Name = "kind-left-cluster"
-	memberCluster6Name = "kind-non-existent-cluster"
+	hubClusterName                = "kind-hub"
+	memberCluster1EastProdName    = "kind-cluster-1"
+	memberCluster2EastCanaryName  = "kind-cluster-2"
+	memberCluster3WestProdName    = "kind-cluster-3"
+	memberCluster4UnhealthyName   = "kind-unhealthy-cluster"
+	memberCluster5LeftName        = "kind-left-cluster"
+	memberCluster6NonExistentName = "kind-non-existent-cluster"
```
Comment (ryanzhang-oss) [inline test/e2e/setup_test.go:45]: nit: do we need the "Name" as a surfix?

File: test/e2e/placement_pickn_test.go [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+							UnavailablePeriodSeconds: pointer.Int(2),
+						},
+					},
+				},
+			}
+			Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP")
+		})
+
+		It("should update CRP status as expected", func() {
+			statusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), []string{memberCluster3WestProdName}, nil)
```
Comment (ryanzhang-oss) [inline test/e2e/placement_pickn_test.go:55]: I wonder why is cluster3 select?

File: test/e2e/placement_pickn_test.go [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+							UnavailablePeriodSeconds: pointer.Int(2),
+						},
+					},
+				},
+			}
+			Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP")
+		})
+
+		It("should place resources on the picked clusters", func() {
+			targetClusters := []*framework.Cluster{memberCluster3WestProd, memberCluster2EastCanary}
```
Comment (ryanzhang-oss) [inline test/e2e/placement_pickn_test.go:102]: same question, why is 2/3 selected? Where is the magic?

File: test/e2e/placement_pickn_test.go [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+						PlacementType:    placementv1beta1.PickNPlacementType,
+						NumberOfClusters: pointer.Int32(2),
+						Affinity: &placementv1beta1.Affinity{
+							ClusterAffinity: &placementv1beta1.ClusterAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: &placementv1beta1.ClusterSelector{
+									ClusterSelectorTerms: []placementv1beta1.ClusterSelectorTerm{
+										{
+											LabelSelector: metav1.LabelSelector{
+												MatchLabels: map[string]string{
+													regionLabelName: regionLabelValue1,
```
Comment (ryanzhang-oss) [inline test/e2e/placement_pickn_test.go:314]: where is the "regionLabelValue1" defined? What is its value?

File: test/e2e/placement_pickn_test.go [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+											},
+										},
+									},
+								},
+							},
+						},
+						TopologySpreadConstraints: []placementv1beta1.TopologySpreadConstraint{
+							{
+								MaxSkew:           pointer.Int32(1),
+								TopologyKey:       envLabelName,
```
Comment (ryanzhang-oss) [inline test/e2e/placement_pickn_test.go:325]: same question, what is this env?

File: test/e2e/setup_test.go (hunk 1 above)
Comment (michaelawyu) [reply test/e2e/setup_test.go:45]: Hi Ryan! The suffix is needed as we also have variables named without the suffix in use (`*framework.Cluster`).

File: test/e2e/placement_pickn_test.go (hunk 2 above)
Comment (michaelawyu) [reply test/e2e/placement_pickn_test.go:55]: Resolved per offline discussion (scheduler will pick by name, i.e., alphanumeric order, when clusters have equal scores).

File: test/e2e/placement_pickn_test.go (hunk 3 above)
Comment (michaelawyu) [reply test/e2e/placement_pickn_test.go:102]: Resolved per offline discussion (scheduler will pick by name, i.e., alphanumeric order, when clusters have equal scores).

File: test/e2e/placement_pickn_test.go (hunk 4 above)
Comment (michaelawyu) [reply test/e2e/placement_pickn_test.go:314]: Hi Ryan! It's in the `setup_test.go` file (line 81 onwards).

File: test/e2e/placement_pickn_test.go (hunk 5 above)
Comment (michaelawyu) [reply test/e2e/placement_pickn_test.go:325]: Same as the comment above.

File: test/e2e/placement_pickn_test.go (hunk 2 above)
Comment (michaelawyu) [reply test/e2e/placement_pickn_test.go:55]: Tweaked the name a little bit as suggested to give clearer hints.

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**